packages/requisitions/src/ResponseRequisition/DetailView/columns.ts

```diff
@@ -1,45 +1,83 @@
+type Store = {
   sortBy: SortBy<ResponseLineFragment>;
-  onChangeSortBy: (
-    column: Column<ResponseLineFragment>
-  ) => SortBy<ResponseLineFragment>;
-}
+  setSortBy: (sortBy: SortBy<ResponseLineFragment>) => void;
+};
 
-export const useResponseColumns = ({
-  onChangeSortBy,
-  sortBy,
-}: UseResponseColumnOptions): Column<ResponseLineFragment>[] =>
-  useColumns<ResponseLineFragment>(
+const useStore = zustand<Store>(set => ({
+  sortBy: { key: 'itemName', isDesc: false, direction: 'asc' },
+  setSortBy: (sortBy: SortBy<ResponseLineFragment>) =>
+    set(state => ({ ...state, sortBy })),
+}));
+
+const useSharedSortBy = () => {
+  const sharedSortBy = useStore();
+  const { sortBy, onChangeSortBy } = useSortBy<ResponseLineFragment>(
+    sharedSortBy.sortBy
+  );
+
+  useEffect(() => {
+    sharedSortBy.setSortBy(sortBy);
+  }, [sortBy]);
+  return { sortBy, onChangeSortBy };
+};
+
+export const useResponseColumns = () => {
+  const { sortBy, onChangeSortBy } = useSharedSortBy();
+  const columns = useColumns<ResponseLineFragment>(
     [
-      ['itemCode', { accessor: ({ rowData }) => rowData.item.code }],
-      ['itemName', { accessor: ({ rowData }) => rowData.item.name }],
-      ['itemUnit', { accessor: ({ rowData }) => rowData.item.unitName }],
       [
-        'stockOnHand',
+        'itemCode',
         {
-          accessor: ({ rowData }) => rowData.itemStats.availableStockOnHand,
-          label: 'label.our-soh',
-          description: 'description.our-soh',
+          accessor: ({ rowData }) => rowData.item.code,
+          getSortValue: rowData => rowData.item.code,
+        },
+      ],
+      [
+        'itemName',
+        {
+          accessor: ({ rowData }) => rowData.item.name,
+          getSortValue: rowData => rowData.item.name,
+        },
+      ],
+      [
+        'itemUnit',
+        {
+          accessor: ({ rowData }) => rowData.item.unitName,
+          getSortValue: rowData => rowData.item.unitName ?? '',
         },
       ],
       [
         'stockOnHand',
         {
-          accessor: ({ rowData }) =>
-            rowData.linkedRequisitionLine?.itemStats?.availableStockOnHand,
-          label: 'label.customer-soh',
-          description: 'description.customer-soh',
+          accessor: ({ rowData }) => rowData.itemStats.availableStockOnHand,
+          getSortValue: rowData => rowData.itemStats.availableStockOnHand,
+          label: 'label.our-soh',
+          description: 'description.our-soh',
         },
       ],
+      {
+        key: 'customerStockOnHand',
+        accessor: ({ rowData }) =>
+          rowData.linkedRequisitionLine?.itemStats?.availableStockOnHand,
+        getSortValue: rowData =>
+          rowData.linkedRequisitionLine?.itemStats?.availableStockOnHand ?? '',
+
+        label: 'label.customer-soh',
+        description: 'description.customer-soh',
+        width: 100,
+        align: ColumnAlign.Right,
+      },
       'requestedQuantity',
```

Comment:
this guy needs a `getSortValue` (I've looked into defaulting `getSortValue` to the `accessor`, but the return types are different)